ravenframework/Optimizers/BayesianOptimizer.py

```diff
@@ -0,0 +1,758 @@
+    trainingSet[self._objectiveVar] = np.asarray(self._trainingTargets[traj])
+    self._model.train(trainingSet)
+    # NOTE It would be preferrable to use targetEvaluation;
+    # however, there does not appear a built in normalization method and as
+    # consequence, it is preferrable to use the in class attributes to train the ROM
+    # on our normalized space.
+    # self._model.train(self._targetEvaluation)
+
+  def _generatePredictiveModel(self, traj):
+    """
+      Applies model selection if necessary, otherwise just fits training data to predictive model
+      @ In, traj, trajectory
+      @ Out, None
+    """
+    # Generate posterior with training data
+    if self._iteration[traj] % self._modelDuration == 0:
+      if self._modelSelection == 'External':
+        self._model.supervisedContainer[0].model.set_params(optimizer='fmin_l_bfgs_b')
+        self._trainRegressionModel(traj)
+        self._model.supervisedContainer[0].model.set_params(optimizer=None)
+      elif self._modelSelection == 'Internal':
+        self._trainRegressionModel(traj)
+        restartCount = self._model.supervisedContainer[0].model.get_params()['n_restarts_optimizer']
+        self._selectHyperparameters(restartCount=restartCount)
+      elif self._modelSelection == 'Average':
+        self.raiseAnError(RuntimeError, 'Model averaging is not yet available')
+      else:
+        self._trainRegressionModel(traj)
+    else:
+      self._trainRegressionModel(traj)
+
+  def _evaluateRegressionModel(self, featurePoint):
+    """
+      Evaluates GPR mean and standard deviation at a given input location
+      @ In, featurePoint, dict, feature values to evaluate ROM
+      @ Out, mu, ROM mean/prediction value at that point
+      @ Out, std, ROM standard-deviation value at that point
+    """
+    # Evaluating the regression model
+    resultsDict = self._model.evaluate(featurePoint)
+    # NOTE only allowing single targets, needs to be fixed when multi-objective optimization is added
+    mu = resultsDict[self._objectiveVar]
+    std = resultsDict[self._objectiveVar+'_std']
+    return mu, std
+
+  # * * * * * * * * * * * *
+  # Constraint Handling
+  def _applyFunctionalConstraints(self, suggested, previous):
+    """
+      Abstract method from RavenSampled, currently not used by BayesianOptimizer
+    """
+    self.raiseAnError(NotImplementedError, '_applyFunctionalConstraints not implemented for Bayesian Optimizer')
+
+  def _handleImplicitConstraints(self, previous):
+    """
+      Considers all implicit constraints
+      @ In, previous, dict, NORMALIZED previous opt point
+      @ Out, accept, bool, whether point was satisfied implicit constraints
+    """
+    normed = copy.deepcopy(previous)
+    oldVal = normed[self._objectiveVar]
+    normed.pop(self._objectiveVar, oldVal)
+    denormed = self.denormalizeData(normed)
+    denormed[self._objectiveVar] = oldVal
+    accept = self._checkImpFunctionalConstraints(denormed)
+    return accept
```

Comment:
This function is the same as the one in RavenSampled, I think you do not need to keep it in this class.